xmtp_id/src/associations/association_log.rs

```diff
@@ -103,6 +109,7 @@ impl IdentityAction for AddAssociation {
     ) -> Result<AssociationState, AssociationError> {
         let existing_state = maybe_existing_state.ok_or(AssociationError::NotCreated)?;
         self.replay_check(&existing_state)?;
+        ensure_matching_inbox_id(&self.inbox_id, existing_state.inbox_id())?;
```

Comment:
Is there a scenario where the wallet is revoked from the current inbox ID, but associated with a different inbox ID, as part of a single batch signature? One of the two will need to be 'skipped over', right?

Reply:
I don't think we want to allow batches that span multiple inboxes. It makes the security model and programming model a lot more complicated, for what's a pretty rare case. Suddenly any update can interfere with any other update. You'd need global total ordering instead of per inbox. And it's not clear what state to apply each action to in advance. 

They can just do two signatures.

Reply:
For the server, I do think we are going to need a top-level `inbox_id` field on the `IdentityUpdate`. Without it, the server is going to have a hard time figuring out what previous state to apply the update to. The way the protos are structured today, the server would need to inspect the individual actions to find the `inbox_id`. 

I'm hoping on the Go side we can be pretty blind to the contents of these Actions and just forward them over to Rust-land for all processing.